java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -851,4 +851,13 @@ public void setInitialCapacity(int valueCount, double density) {
         }
       }
     }
+
+  /**
+   * Set the element at the given index to null. For DenseUnionVector, it is a no-op as nulls are not supported at the
+   * top level, and isNull() always returns false. Nullability is only handled at the level of individual child vectors.
+   *
+   * @param index position of element
+   */
+  @Override
+  public void setNull(int index) {}
```

Comment:
Same here: should this raise an error?

Reply:
Or left unimplemented/abstract if possible

Reply:
It needs to be implemented (or throw an exception) to make it possible to use myArbitraryFieldVector.setNull() without casting for all the vector types